content/ConnectingToDatabase/ConnectingToDatabase.q.md

```diff
@@ -0,0 +1,837 @@
+
+```php
+<?php
+    $host = 'newhost';
+    $db = 'MyDatabase';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected!"; 
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection.";
+    }
+?>
+```
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+   $host = 'localhost';
+   $db = 'MyDatabase';
+   $port = '5432';
+   $username = 'postgres';
+   $password = 'Admin01';
+   
+   $dsn = "pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password";
+
+   try {
+       $conn = new PDO($dsn);
+       if ($conn) {
+           echo "Successfully connected!";
+       }
+   } catch (Exception $e) {
+       echo "Unable to establish a connection.";
+   }
+?>
+```
+/// type=SS, answer=[3]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Successfully connected!`.
+
+- It prints `"Successfully connected!"`.
+
+- It prints `Unable to establish a connection.`.
+
+
+/// type=SS, answer=[4]
+
+On line 4, replace the string value `5432` with `1234`. Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Successfully connected!`.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `"Unable to establish a connection."`.
+
+
++++
+
+
++++
+
+### Part 4: Practice
+
+/// type=CR, answer=[tests/ConnectingToDatabase/CreatePHPApplicationConnectingToPostgreSQLTest.php]
+
+Write a PHP program that uses the `PDO_PGSQL` driver, `PDO_PGSQL DSN`, and `PDO` class to connect to the PostgreSQL database. First, create the following variables `$host`, `$db`, `$port`, `$username`, and `$password`. Assign the values `localhost`, `MyDatabase`, `5432`, `postgres`, and `Admin01` respectively to the variables. Then, assign the `PDO_PGSQL DSN` statement which contains the DSN parameters `host`, `port`, `dbname`, `username`, and `password` to the `$dsn` variable. Set the DSN parameters with their respective variables `$host`, `$db`, `$port`, `$username`, and `$password`. Next, create the `try` block which contains two statements. In the first statement, create the `$conn` object an instance of the `PDO` class which passes the argument `$dsn`. In the second statement, write the `if` statement to evaluate the `$conn` object inside the parentheses `()`. If the expression evaluates to `true` then, the `echo` statement displays the string `Successfully connected!`. Set the `catch` block to retrieve the exception if an exception occurs within the `try` block. Create the `$e` object an instance of the `Exception` class. Then, display the error message `Unable to establish a connection.` inside the `catch` block. Run the program to view the output.
```

Comment:
Please take note of this statement __Set the `catch` block to retrieve the exception if an exception occurs within the `try` block. Create the `$e` object an instance of the `Exception` class.__ as you made changes to the definition of `catch` in our facts. Thanks.